FutureNorthants/northants-website
# Added page_titles query to graphql API

### Testing

1. With this branch checked out and your backend up and running, go to https://core-cms.northants.localhost/graphql - expect page to display as normal. In the documentation explorer on the right, verify you can click through on the schema (Query -> page_types -> PageType) to see how the new query is constructed.
2. Try entering this query in the query window: `{ page_titles }` and clicking the execute query (play) button. Verify you get a load of results.
3. Try filtering the query by a piece of text: `{ page_titles(containing: "park") }` and verify that the results have been filtered accordingly.
4. Try filtering the query by a page type: `{ page_titles(page_types: PRESSRELEASE) }` and verify that the results have been filtered accordingly. The other option for page_types is ARTICLE (note no quotes are needed around these values, and you can combine both by wrapping them in square brackets).
5. Try combining the two filters e.g. `{ page_titles(containing: "2021", page_types: PRESSRELEASE) }` and verify both filters are applied accordingly.
6. Try entering this query in the query window: `{ route(path: "/") { node { title, searchSuggestions }}}` and verify that valid titles come back in the searchSuggestions field.
7. Try other pages in the same query e.g. `path : "/your-council/contact-us"` and `path: "/news/test-twice-weekly-curb-spread-covid-19"` and verify they come back with searchSuggestions populated too, regardless of whether they are service landing, service or news article pages.
